src/lib/parser.c

```diff
@@ -105,90 +155,234 @@ parser_error_t parser_getItem(const parser_context_t *ctx,
-            err = parser_getAddress(parser_tx_obj.chainID, parser_tx_obj.chainIDLen,
-                                    (char *) UI_buffer, UI_BUFFER,
-                                    parser_tx_obj.sendmsg.sourcePtr,
-                                    parser_tx_obj.sendmsg.sourceLen);
-            // page it
-            parser_arrayToString(outValue, outValueLen, UI_buffer,
-                                 strlen((char *) UI_buffer), pageIdx, pageCount);
+        }
+        case Msg_Vote: {
+            switch (parser_mapDisplayIdx(ctx, displayIdx)) {
+                case FIELD_CHAINID:     // ChainID
+                    snprintf(outKey, outKeyLen, "ChainID");
+                    err = parser_arrayToString(outValue, outValueLen,
+                                               parser_tx_obj.chainID,
+                                               parser_tx_obj.chainIDLen,
+                                               pageIdx, pageCount);
+                    break;
+                case FIELD_VOTER: {     // Voter
+                    snprintf(outKey, outKeyLen, "Voter");
+                    err = parser_getAddress(parser_tx_obj.chainID, parser_tx_obj.chainIDLen,
+                                            (char *) UI_buffer, UI_BUFFER,
+                                            parser_tx_obj.votemsg.voterPtr,
+                                            parser_tx_obj.votemsg.voterLen);
+                    // page it
+                    parser_arrayToString(outValue, outValueLen, UI_buffer,
+                                         strlen((char *) UI_buffer), pageIdx, pageCount);
+                    break;
+                }
+                case FIELD_PROPOSAL_ID: { //Proposal Id
+                    snprintf(outKey, outKeyLen, "ProposalId");
+                    uint8_t bcdOut[20]; //Must be  at most outValueLen/2
+                    uint16_t bcdOutLen = sizeof(bcdOut);
+                    bignumBigEndian_to_bcd(bcdOut, bcdOutLen, parser_tx_obj.votemsg.proposalIdPtr,
+                                           parser_tx_obj.votemsg.proposalIdLen);
+                    bignumBigEndian_bcdprint(outValue, outValueLen, bcdOut, bcdOutLen);
+                    break;
+                }
+                case FIELD_SELECTION: { // Vote option
+                    char *sel;
+                    switch (parser_tx_obj.votemsg.voteOption) {
+                        case VOTE_OPTION_YES:
+                            sel = VOTE_OPTION_YES_STR;
+                            break;
+                        case VOTE_OPTION_NO:
+                            sel = VOTE_OPTION_NO_STR;
+                            break;
+                        case VOTE_OPTION_ABSTAIN:
+                            sel = VOTE_OPTION_ABSTAIN_STR;
+                            break;
+                        default:
+                            sel = VOTE_OPTION_INVALID_STR;
+                    }
+                    MEMCPY(outValue, sel, sizeof(sel));
```

Comment:
sizeof(sel) is actually the size of the pointer.. not the string.
it would be better if you use snprintf to be sure we don't write beyond the output buffer

Reply:
Replaced for:
`snprintf(outValue, outValueLen, "%s", sel);`